src/pal/inc/pal.h

```diff
@@ -201,7 +201,13 @@ extern "C" {
 
 #else // _MSC_VER
 
+#if defined (_M_ARM) && !defined (DEBUG)
 #define UNALIGNED
+#define UNALIGNED_ARM     __attribute__((aligned(1))) //for -O2 and -O3
```

Comment:
Can you just define the UNALIGNED to __attribute__((aligned(1))) instead of introducing the UNALIGNED_ARM?